integ-news/service/src/main/java/org/exoplatform/news/NewsServiceImpl.java

```diff
@@ -95,10 +107,20 @@ public NewsServiceImpl(RepositoryService repositoryService, SessionProviderServi
    * @throws RepositoryException
    */
   public News createNews(News news) throws Exception {
-    String newsId = createNewsNode(news);
-
-    news.setId(newsId);
+    SessionProvider sessionProvider = sessionProviderService.getSessionProvider(null);
+    Session session = sessionProvider.getSession(repositoryService.getCurrentRepository().getConfiguration().getDefaultWorkspaceName(),
+        repositoryService.getCurrentRepository());
 
+    updateNews(news);
+    try {
+      Node newsNode = session.getNodeByUUID(news.getId());
+      HashMap<String, String> context = new HashMap<String, String>();
+      this.publicationServiceImpl.changeState(newsNode, "published", context);
```

Comment:
```suggestion
      this.publicationServiceImpl.changeState(newsNode, "published", new HashMap<String, String>());
```
and remove the line above.

Reply:
okay